src/core/component.js

```diff
@@ -1,10 +1,22 @@
+/* global HTMLElement */
+var schema = require('./schema');
+var coordinates = require('../utils/coordinates');
 var debug = require('../utils/debug');
 var styleParser = require('style-attr');
 var utils = require('../utils/');
 
-var components = module.exports.components = {};  // Keep track of registered components.
+var coerceToSchema = schema.coerce;
+var coerceValueToSchema = schema.coerceValue;
+var processSchema = schema.process;
 var error = debug('core:register-component:error');
 
+var components = module.exports.components = {};  // Keep track of registered components.
+var types = {};  // Registered component types.
+
+registerComponentType('boolean', booleanParse, booleanStringify);
```

Comment:
This is different from property types, which I have also implemented in this PR.

This is for top-level parsing for the whole component data rather than parsing on an individual property.